### PR TITLE
Add libidn2

### PR DESCRIPTION
**Pull request by hmaarrfk**
Not sure if this is needed for gnutls https://github.com/conda-forge/gnutls-feedstock/pull/9
But maybe, and I want to test if it works.

@chenghlee do you want to be a maintainer here?

cc: @mariusvniekerk @scopatz

### CODE REVIEW

File: recipes/libidn2/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    #    https://abi-laboratory.pro/tracker/timeline/libidn2/
+    - {{ pin_subpackage('libidn2', min_pin='x', max_pin='x') }}
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+
+test:
+  commands:
+    - test -f "${PREFIX}/include/idn2.h"
```
Comment (carlodri) [inline recipes/libidn2/meta.yaml]:
```suggestion
    - if not exist "%PREFIX%\include\idn2.h" exit 1  # [win]
```
maybe better

File: recipes/libidn2/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    - test -f "${PREFIX}/lib/libidn2.so"     # [linux]
+    - test -f "${PREFIX}/lib/libidn2.dylib"  # [osx]
+    - conda inspect linkages -p "$PREFIX" "$PKG_NAME"
```
Comment (carlodri) [inline recipes/libidn2/meta.yaml]:
```suggestion
    - conda inspect linkages -p "$PREFIX" "$PKG_NAME"  # [unix]
```

AFAIK works only on mac an linux

**Comment by hmaarrfk** [comment]
@carlodri thanks for the review. I really don't know how to build things correctly on windows. Seeing as this is probably only needed for gnutls, which isn't built for windows, I think it is OK.

**Comment by hmaarrfk** [comment]
Also, maybe lets let somebody working on the gnutls feedstock merge this is. I don't know if it is necessary, but I just put it out there in case. I would rather not orphan this feedstock later.

**Comment by scopatz** [comment]
Thanks!

**Pull request merged**